moveit_ros/occupancy_map_monitor/src/occupancy_map_monitor.cpp

```diff
@@ -39,124 +39,95 @@
+namespace
 {
-  initialize();
+static const rclcpp::Logger LOGGER = rclcpp::get_logger("moveit.ros.occupancy_map_monitor");
 }
 
-OccupancyMapMonitor::OccupancyMapMonitor(const rclcpp::Node::SharedPtr& node,
+OccupancyMapMonitor::OccupancyMapMonitor(std::unique_ptr<MiddlewareHandle> middleware_handle,
                                          const std::shared_ptr<tf2_ros::Buffer>& tf_buffer,
                                          const std::string& map_frame, double map_resolution)
-  : tf_buffer_(tf_buffer)
+  : middleware_handle_(std::move(middleware_handle))
+  , tf_buffer_(tf_buffer)
   , map_frame_(map_frame)
   , map_resolution_(map_resolution)
   , debug_info_(false)
   , mesh_handle_count_(0)
-  , node_(node)
+  , active_(false)
 {
   initialize();
 }
 
+OccupancyMapMonitor::OccupancyMapMonitor(const rclcpp::Node::SharedPtr& node, double map_resolution)
+  : OccupancyMapMonitor(std::make_unique<OccupancyMapMonitorMiddlewareHandle>(node), nullptr, "", map_resolution)
+{
+}
+
+OccupancyMapMonitor::OccupancyMapMonitor(const rclcpp::Node::SharedPtr& node,
+                                         const std::shared_ptr<tf2_ros::Buffer>& tf_buffer,
+                                         const std::string& map_frame, double map_resolution)
+  : OccupancyMapMonitor(std::make_unique<OccupancyMapMonitorMiddlewareHandle>(node), tf_buffer, map_frame,
+                        map_resolution)
+{
+}
+
 void OccupancyMapMonitor::initialize()
 {
-  /* load params from param server */
-  if (map_resolution_ <= std::numeric_limits<double>::epsilon())
-    if (!node_->get_parameter("octomap_resolution", map_resolution_))
-    {
-      map_resolution_ = 0.1;
-      RCLCPP_WARN(LOGGER, "Resolution not specified for Octomap. Assuming resolution = %g instead", map_resolution_);
-    }
+  map_resolution_ = middleware_handle_->getMapResolutionParameter();
   RCLCPP_DEBUG(LOGGER, "Using resolution = %lf m for building octomap", map_resolution_);
 
   if (map_frame_.empty())
-    if (!node_->get_parameter("octomap_frame", map_frame_))
-      if (tf_buffer_)
-        RCLCPP_WARN(LOGGER, "No target frame specified for Octomap. No transforms will be applied to received data.");
+  {
+    map_frame_ = middleware_handle_->getMapFrameParameter();
+  }
 
+  if (tf_buffer_ && map_frame_.empty())
+  {
+    RCLCPP_WARN(LOGGER, "No target frame specified for Octomap. No transforms will be applied to received data.");
+  }
   if (!tf_buffer_ && !map_frame_.empty())
+  {
     RCLCPP_WARN(LOGGER, "Target frame specified but no TF instance (buffer) specified."
                         "No transforms will be applied to received data.");
+  }
 
-  tree_.reset(new OccMapTree(map_resolution_));
+  tree_ = std::make_shared<OccMapTree>(map_resolution_);
   tree_const_ = tree_;
 
-  std::vector<std::string> sensor_list;
-  if (node_->get_parameter("sensors", sensor_list))
+  std::vector<std::string> sensor_list = middleware_handle_->getSensorListParameter();
```

Comment:
`const`?

Reply:
sure, wouldn't hurt here

Reply:
https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rconst-const

I would argue that it always hurts when a scope variable that can be `const`, isn't.